softwareComponents/parsing/include/parsing/parsing.hpp

```diff
@@ -0,0 +1,189 @@
+#pragma once
+
+#include <istream>
+#include <ostream>
+#include <span>
+#include <vector>
+
+#include <atoms/result.hpp>
+#include <atoms/unreachable.hpp>
+#include <configuration/rofiworld.hpp>
+#include <configuration/serialization.hpp>
+
+#include <voxel.hpp>
+
+#include "parsing/parsing_lite.hpp"
+
+
+namespace rofi::parsing
+{
+namespace detail
+{
+    inline void printJson( std::ostream & ostr, const nlohmann::json & json )
+    {
+        ostr.width( 4 );
+        ostr << json << std::endl;
+    }
+} // namespace detail
+
+
+enum class RofiWorldFormat {
+    Old = -1,
+    Json,
+    Voxel,
+};
+
+inline auto operator<<( std::ostream & ostr, RofiWorldFormat worldFormat ) -> std::ostream &
+{
+    switch ( worldFormat ) {
+        case RofiWorldFormat::Old:
+            return ostr << "old";
+        case RofiWorldFormat::Json:
+            return ostr << "json";
+        case RofiWorldFormat::Voxel:
+            return ostr << "voxel";
+    }
+    assert( false );
+    return ostr << static_cast< int >( worldFormat );
```

Comment:
Tohle je nedosažitelný kód.

Reply:
Chtěl jsem aby v release to vypsalo správně hodnotu (když přijde něco, co není možnost v enumu), protože nevím jak moc to jde v dimcli obejít, ale asi je pravda, že to je spíš chyba programátora v takovém případě (a jak se to používá teď se to stát nemůže).